### PR TITLE
Added moq warning include, and place it near current references.

### PR DESCRIPTION
**Pull request by IEvangelist**
## Summary

Add Moq warning.

Related to #36629

---

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [docs/architecture/microservices/multi-container-microservice-net-applications/test-aspnet-core-services-web-apps.md](https://github.com/dotnet/docs/blob/7282d8ba0cafe9e0caf179e385d9eaca3e531373/docs/architecture/microservices/multi-container-microservice-net-applications/test-aspnet-core-services-web-apps.md) | [Testing ASP.NET Core services and web apps](https://review.learn.microsoft.com/en-us/dotnet/architecture/microservices/multi-container-microservice-net-applications/test-aspnet-core-services-web-apps?branch=pr-en-us-36638) |
| [docs/azure/sdk/unit-testing-mocking.md](https://github.com/dotnet/docs/blob/7282d8ba0cafe9e0caf179e385d9eaca3e531373/docs/azure/sdk/unit-testing-mocking.md) | [Unit testing and mocking with the Azure SDK for .NET](https://review.learn.microsoft.com/en-us/dotnet/azure/sdk/unit-testing-mocking?branch=pr-en-us-36638) |
| [docs/core/deploying/runtime-store.md](https://github.com/dotnet/docs/blob/7282d8ba0cafe9e0caf179e385d9eaca3e531373/docs/core/deploying/runtime-store.md) | [Runtime package store](https://review.learn.microsoft.com/en-us/dotnet/core/deploying/runtime-store?branch=pr-en-us-36638) |
| [docs/csharp/advanced-topics/expression-trees/index.md](https://github.com/dotnet/docs/blob/7282d8ba0cafe9e0caf179e385d9eaca3e531373/docs/csharp/advanced-topics/expression-trees/index.md) | [Expression Trees](https://review.learn.microsoft.com/en-us/dotnet/csharp/advanced-topics/expression-trees/index?branch=pr-en-us-36638) |
| [docs/orleans/implementation/testing.md](https://github.com/dotnet/docs/blob/7282d8ba0cafe9e0caf179e385d9eaca3e531373/docs/orleans/implementation/testing.md) | [docs/orleans/implementation/testing](https://review.learn.microsoft.com/en-us/dotnet/orleans/implementation/testing?branch=pr-en-us-36638) |
| [docs/orleans/resources/best-practices.md](https://github.com/dotnet/docs/blob/7282d8ba0cafe9e0caf179e385d9eaca3e531373/docs/orleans/resources/best-practices.md) | [Best practices in Orleans](https://review.learn.microsoft.com/en-us/dotnet/orleans/resources/best-practices?branch=pr-en-us-36638) |

### CODE REVIEW

**Review by gnalvesteffer — APPROVED** [review]

File: docs/architecture/microservices/multi-container-microservice-net-applications/test-aspnet-core-services-web-apps.md [hunk 1]
```diff
@@ -28,6 +28,8 @@ As you unit test your controller actions, make sure you focus only on their beha
 
 Unit tests are implemented based on test frameworks like xUnit.net, MSTest, Moq, or NUnit. For the eShopOnContainers sample application, we are using xUnit.
```
Comment (jsquire) [inline docs/architecture/microservices/multi-container-microservice-net-applications/test-aspnet-core-services-web-apps.md:29]: nit:  Moq is not a test framework and probably shouldn't be in this list.

**Review by jsquire — APPROVED** [review]
Azure SDK bits and the warning LGTM.

File: docs/architecture/microservices/multi-container-microservice-net-applications/test-aspnet-core-services-web-apps.md (hunk 1 above)
Comment (IEvangelist) [inline docs/architecture/microservices/multi-container-microservice-net-applications/test-aspnet-core-services-web-apps.md:29]:
```suggestion
Unit tests are implemented based on test frameworks like xUnit.net, MSTest, or NUnit. For the eShopOnContainers sample application, we are using xUnit.
```

File: docs/architecture/microservices/multi-container-microservice-net-applications/test-aspnet-core-services-web-apps.md [hunk 2]
```diff
@@ -28,6 +28,8 @@ As you unit test your controller actions, make sure you focus only on their beha
...
 
+[!INCLUDE [moq-warning](../../../../includes/moq-warning.md)]
+
```
Comment (IEvangelist) [inline docs/architecture/microservices/multi-container-microservice-net-applications/test-aspnet-core-services-web-apps.md:32]:
```suggestion
```

File: includes/moq-warning.md [hunk 3]
```diff
@@ -0,0 +1,2 @@
+> [!WARNING]
+> We no longer recommend [Moq](https://github.com/moq/moq) for new projects. Existing Moq users should consider [a recent issue posted](https://github.com/moq/moq/issues/1372) concerning changes from its maintainers for context. Moq content may be removed at a later date.
```
Comment (richlander) [inline includes/moq-warning.md:2]: We should likely be a bit more direct.

```suggestion
> We no longer recommend [Moq](https://github.com/moq/moq) for new projects. Existing Moq users should consider recent [privacy-related changes](https://github.com/moq/moq/issues/1372) to the library. Moq content may be removed at a later date.
```

Comment (gewarren) [reply includes/moq-warning.md:2]:
```suggestion
> Due to recent [privacy-related changes](https://github.com/moq/moq/issues/1372) in the [Moq](https://github.com/moq/moq) library, we don't recommend it for new projects. If you're using Moq in an existing project, you should carefully consider the risks. Moq content might be removed from learn.microsoft.com at a later date.
```

**Review by scottaddie — APPROVED** [review]

**Review by BillWagner — APPROVED** [review]
This LGTM @IEvangelist 

Let's :shipit: once the product team is happy with the additions.

**Comment by IEvangelist** [comment]
Closing for now per a conversation with @richlander.

**Pull request closed without merging**